### ISSUE TITLE
support template for extracting date and time information 

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/iobox-win32#7]
For example: YYYY means the year component of the date. 

For GPCR use case, we just need the year of the last_modified timestamp of the uploaded file.

**Comment by karlcz** [comment]
We should add a date_parse filter that can be run as another rule step to take an input date by template and convert it to output fields.  We should take a look at existing date formatting tools in Python.  Since we have to support win32, we cannot just rely on the GNU `date` utility which can do all kinds of parsing and reformatting...

**Comment by svoinea** [comment]
Created the **datetime** handler that converts a timestamp defined in the `input` section from one format to another one defined in the `output` section. The input timestamp string is defined in the `date_string` parameter and its format in the `format` parameter. The `output` section contains the `format` to convert to. Example:

```
      {
         "handler": "datetime",
         "input": {
                  "date_string": "2016-05-06 20:39:08.982442",
                  "format": "%Y-%m-%d %H:%M:%S.%f"
               },
         "output": {
                  "date": {
                              "format": "%Y-%m-%d"
                          }
                }
      }

```

**Comment by karlcz** [comment]
I think the output section content should just be:

```
"key": "format string", ...
```

Rather than:

```
"key": { "format": "format string" }, ...
```

**Comment by svoinea** [comment]
Fixed through commit e93d7aa34a2e5d5419b5172d5f05390eb1ab366b.
